### PR TITLE
Rewrite resources to remove Poise dependency

### PR DESCRIPTION
**Pull request by josqu4red**

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -19,10 +19,11 @@
 # build-essential is obsolete in chef 14+
 depends 'build-essential' # ~FC121
 depends 'nssm', '>= 3.42.0'
-depends 'golang'
+depends 'golang', '< 5.0.0'
```
Comment (jeremy-clerc) [inline metadata.rb]: < constraints should not be added here, if it breaks the tests, add it to Berksfile.
Else this cookbook may be compatible with Chef > 14 while we restrict golang for Chef 14.

File: metadata.rb [hunk 2]
```diff
@@ -20,9 +20,7 @@
 depends 'build-essential' # ~FC121
 depends 'nssm', '>= 3.42.0'
 depends 'golang', '< 5.0.0'
-depends 'poise', '~> 2.2'
 depends 'poise-archive', '~> 1.3'
```
Comment (jeremy-clerc) [inline metadata.rb]: Not poise-archive too ?

File: resources/execute.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+default_action(:run)
```
Comment (jeremy-clerc) [inline resources/execute.rb]: NIT: Remove parentheses here, for property and action to be consistent with other resources

File: resources/watch.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+default_action(:create)
```
Comment (jeremy-clerc) [inline resources/watch.rb]: NIT: Here too remove parantheses, or use parenthenses in other files, for consistency

File: resources/config_v0.rb [hunk 5]
```diff
@@ -0,0 +1,269 @@
+provides :consul_config do |node|
```
Comment (jeremy-clerc) [inline resources/config_v0.rb:1]: OPT: consul < 1 is 3 years old, IMHO we could drop support.

File: resources/service.rb [hunk 6]
```diff
@@ -0,0 +1,119 @@
+# @!property config_file
```
Comment (jeremy-clerc) [inline resources/service.rb:1]: OPT: As you drop upstart / sysvinit service config, add it to commit message.

File: resources/installation.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+action :create do
+  attrs = node['consul']['install']['binary']
+
+  directory ::File.dirname(consul_program) do
+    mode '0755'
+    recursive true
+  end
+
+#  ark 'consul' do # TODO: replace with archive_file when Chef>15
```
Comment (jeremy-clerc) [inline resources/installation.rb]: Cleanup commented code

File: resources/installation.rb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    strip_components 0
+    not_if { ::File.exist?(consul_program) }
+  end
+
+  link '/usr/local/bin/consul' do
+    to consul_program
+    not_if { node.windows? }
+  end
+
+#  link "#{node.config_prefix_path}\\consul.exe" do
```
Comment (jeremy-clerc) [inline resources/installation.rb]: Cleanup here too

File: resources/installation.rb (hunk 7 above)
Comment (jeremy-clerc) [reply resources/installation.rb]: Or is it supposed to be used ?

File: attributes/default.rb [hunk 9]
```diff
@@ -8,6 +8,7 @@
 default['consul']['service_name'] = 'consul'
 default['consul']['service_user'] = 'consul'
 default['consul']['service_group'] = 'consul'
+default['consul']['service_shell'] = '/bin/false'
```
Comment (jeremy-clerc) [inline attributes/default.rb:11]: Default via poise ?

File: recipes/default.rb [hunk 10]
```diff
@@ -4,9 +4,18 @@
...
+  not_if { node.platform_family?('windows') }
+  not_if { node['consul']['service_group'] == 'root' }
+  not_if { node['consul']['create_service_user'] == false }
+end
+
+user node['consul']['service_user'] do
+  system true
   group node['consul']['service_group']
-  shell node['consul']['service_shell'] unless node['consul']['service_shell'].nil?
+  shell node['consul']['service_shell']
```
Comment (jeremy-clerc) [inline recipes/default.rb:18]: Linked to other comment, if default via poise. If not, why force the shell ?

**Review by jeremy-clerc — CHANGES_REQUESTED** [review]

File: attributes/default.rb (hunk 9 above)
Comment (josqu4red) [reply attributes/default.rb:11]: `nologin` is preferred but it defaults to `false`.
https://github.com/poise/poise-service/blob/master/lib/poise_service/resources/poise_service_user.rb#L86

File: resources/installation.rb (hunk 7 above)
Comment (josqu4red) [reply resources/installation.rb]: Done

File: resources/installation.rb (hunk 8 above)
Comment (josqu4red) [reply resources/installation.rb]: Done

File: metadata.rb (hunk 2 above)
Comment (josqu4red) [reply metadata.rb]: I hoped that poise-archive did not rely on the problematic lib, but it is :(

File: recipes/default.rb (hunk 10 above)
Comment (josqu4red) [reply recipes/default.rb:18]: Not sure to understand, but if `service_shell` is nil, shell will be nil anyways, so guard is useless

**Review by jeremy-clerc — APPROVED** [review]

**Pull request merged**